pkg/fanal/secret/builtin-rules.go

```diff
@@ -783,4 +784,12 @@ var builtinRules = []Rule{
 		SecretGroupName: "secret",
 		Keywords:        []string{"typeform"},
 	},
+	{
+		ID:              "azure-data-factory-registration-key",
+		Category:        CategoryAzure,
+		Title:           "Azure Data Factory Self-Hosted Integration Runtime registration key",
+		Severity:        "HIGH",
+		Regex:           MustCompile(`IR@[0-9a-zA-Z-]{36}@(.*?)@[0-9a-zA-Z]*@\S+`),
```

Comment:
> the one my PR is based on was a real one from an ADF resource that I own

Have you worked with only one of your secrets?
Can you create some new test secrets and check length of Base64 string?

Better if we will use exact length to avoid case when we take extra characters in secret.

And one more moment - base64 only uses `A-Za-z0-9+/`.


```suggestion
		Regex:           MustCompile(`IR@[0-9a-zA-Z-]{36}@(.*?)@[0-9a-zA-Z]*@[A-Za-z0-9+\/]{26}`),
```